src/vs/workbench/electron-browser/window.ts

```diff
@@ -415,25 +416,17 @@ export class NativeWindow extends Disposable {
 			}
 		});
 
+		// Edit argv.json with crash reporter configs for a fresh start.
+		if (this.environmentService.createCrashReporterConfig) {
```

Comment:
@deepak1556 I would actually move this into `electron-main` because if a user opens multiple windows at once, they might all fight over this in parallel. 

Since `IConfigurationService` can be accessed in `electron-main` too, [`afterWindowOpen`](https://github.com/microsoft/vscode/blob/bbe787ac280cab75002b71460a93ab725d810e80/src/vs/code/electron-main/app.ts#L805) would be a good fit imho since this can happen after windows opened.

One downside is that I do not see the json editing service available in `electron-main`, but the primitives for writing should be available: https://github.com/microsoft/vscode/blob/bbe787ac280cab75002b71460a93ab725d810e80/src/vs/base/common/jsonEdit.ts#L1